stvisualizer/packages/vs-syntax-tree/src/syntaxTree.tsx

```diff
@@ -0,0 +1,140 @@
+"use strict";
+/**
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+import React, { useEffect, useState } from "react";
+import { Button, Dimmer, Icon, Label, Loader } from "semantic-ui-react";
+import DropdownTree from "./representations/dropdown-tree";
+import GraphicalTree from "./representations/graphical-tree";
+import { ERROR_MESSAGE, FULL_TREE_MODE, SWITCH_DROPDOWN, SWITCH_GRAPHICAL } from "./resources/constants";
+import { PrimaryProps, TreeGraph, TreeNodeObject } from "./resources/tree-interfaces";
+import * as styles from "./styles/primary.styles";
+
+function SyntaxTree(props: PrimaryProps) {
+    const {activatedCommand, onFindNode, onCollapseTree, renderTree, switchFullTree} = props;
+
+    const [responseStatus, setResponseStatus] = useState(true);
+    const [isDropdownView, setIsDropdownView] = useState(false);
+    const [hoverViewSwitch, setViewSwitchStatus] = useState(false);
+    const [hoverFullTreeSwitch, setFullTreeSwitchStatus] = useState(false);
+    const [graphicalTree, setGraphicalTree] = useState<TreeGraph | undefined>(undefined);
+    const [dropdownTree, setDropdownTree] = useState<TreeNodeObject[] | undefined>(undefined);
+
+    useEffect(() => {
+        renderTree().then((result) => {
+            if (result.treeArray && result.treeGraph) {
+                setResponseStatus(true);
+                setGraphicalTree(result.treeGraph);
+                setDropdownTree(result.treeArray);
+            } else {
+                setResponseStatus(false);
+            }
+        });
+    }, [props]);
```

Comment:
I assume you want to run this function only once (at `componentDidMount` lifecycle event). If so, we can keep the dependencies array empty (`[]`). There's no point of adding the `props` here since it will have the same effect as empty array of dependencies. Therefore, let's use `[]`?

Reply:
No, not only on mount. This needs to be triggered when collapsing the tree as well.

Reply:
What's the property you are using to determine whether the tree was collapsed? Or is it coming from a parent component?